### DISCUSSION TITLE
got装饰器的prompt用法

### DISCUSSION THREAD

**Discussion by Dragon-ts** [nonebot/discussions#427] [问答]
我用了`matcher.got(key, prompt)`装饰器，但是现在我想在`prompt`中发送如`MessageSegment.at()`的信息，其中at有参数为user_id，我怎么才能获取到响应者的user_id？
我尝试在`matcher.handle()`中添加了`event: Event`，但是仍然无法完成，请问有什么解决的办法
如有表达不清请见谅

**Comment by yanyongyu** [comment]
使用MessageTemplate，got的prompt参数与send的消息参数一致，可以使用消息模板

**Reply by Dragon-ts** [reply]
我看了下关于Message.template()的文档，简单写了一下

```py
test = on_command("test")

@start.handle()
async def _(matcher: Matcher, event: GroupMessageEvent, state: T_State, args: Message = CommandArg()):
	plain_text = args.extract_plain_text()
	state['a'] = 1

@throw.got("next", prompt=Message.template("{a}"))
async def handled_(event: GroupMessageEvent):
        pass
```

但是发现变量a并不能传递过去，可能是我理解有误，麻烦您指点一下，谢谢！

**Reply by yanyongyu** [reply]
请提供一下debug日志看下，另外确保你的Message类是从适配器倒入的。

以及你的`test`响应器为什么下面是，`@start.handle()` `@throw.got()`
